src/db/migrations/1762074152438-AddPartsValidations.ts

```diff
@@ -0,0 +1,484 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class AddPartsValidations1762074152438 implements MigrationInterface {
+    name = 'AddPartsValidations1762074152438'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
+      CREATE TABLE IF NOT EXISTS "polygon_parts"."base_parts" (
+          "product_id"               text COLLATE "ucs_basic" NOT NULL,
+          "product_type"             "polygon_parts"."product_type_enum" NOT NULL,
+          "source_id"                text COLLATE "ucs_basic",
+          "source_name"              text COLLATE "ucs_basic" NOT NULL,
+          "product_version"          text COLLATE "ucs_basic" NOT NULL,
+          "ingestion_date_utc"       TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+          "imaging_time_begin_utc"   TIMESTAMP WITH TIME ZONE NOT NULL,
+          "imaging_time_end_utc"     TIMESTAMP WITH TIME ZONE NOT NULL,
+          "resolution_degree"        numeric NOT NULL,
+          "resolution_meter"         numeric NOT NULL,
+          "source_resolution_meter"  numeric NOT NULL,
+          "horizontal_accuracy_ce90" numeric NOT NULL,
+          "sensors"                  text COLLATE "ucs_basic" NOT NULL,
+          "countries"                text COLLATE "ucs_basic",
+          "cities"                   text COLLATE "ucs_basic",
+          "description"              text COLLATE "ucs_basic",
+          "id"                       text NOT NULL,
+          "catalog_id"               uuid NOT NULL,
+          CONSTRAINT "base_product id"
+              CHECK ("product_id" ~ '^[A-Za-z]{1}[A-Za-z0-9_]{0,37}$'),
+          CONSTRAINT "base_product version"
+              CHECK ("product_version" ~ '^[1-9]\\d*(\\.(0|[1-9]\\d?))?$'),
+          CONSTRAINT "base_imaging time begin utc"
+              CHECK ("imaging_time_begin_utc" < now()),
+          CONSTRAINT "base_imaging time end utc"
+              CHECK ("imaging_time_end_utc" < now()),
+          CONSTRAINT "imaging times" CHECK (
+              "imaging_time_begin_utc" <= "imaging_time_end_utc"),
+          CONSTRAINT "base_resolution degree"
+              CHECK ("resolution_degree" BETWEEN 0.000000167638063430786 AND 0.703125),
+          CONSTRAINT "base_resolution meter"
+              CHECK ("resolution_meter" BETWEEN 0.0185 AND 78271.52),
+          CONSTRAINT "base_source resolution meter"
+              CHECK ("source_resolution_meter" BETWEEN 0.0185 AND 78271.52),
+          CONSTRAINT "base_horizontal accuracy ce90"
+              CHECK ("horizontal_accuracy_ce90" BETWEEN 0.01 AND 4000),
+          CONSTRAINT "base_parts_pkey" PRIMARY KEY ("id")
+      );
+    `);
+
+    // --- Base indexes ---
```

Comment:
if the table serves as a template for others index defition can be dropped since indices are not shared to inherited tables